File: backend/src/main/java/ropold/backend/model/CustomerModel.java
package ropold.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "customers")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CustomerModel {

    @Id
    private UUID id;

    @Column(name = "debitor_nr_navision", length = 50)
    private String debitorNrNavision;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "contact_person")
    private String contactPerson;

    @Column(name = "contact_details", columnDefinition = "TEXT")
    private String contactDetails;

    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(name = "is_archived", nullable = false, columnDefinition = "BOOLEAN DEFAULT false")
    private Boolean isArchived = false;
}

File: frontend/src/components/model/CustomerModel.ts

export type CustomerModel = {
    id: string;
    debitorNrNavision?: string;
    name: string;
    contactPerson?: string;
    contactDetails?: string;
    notes?: string;
    imageUrl?: string;
    isArchived?: boolean;
}

export const DefaultCustomer: CustomerModel = {
    id: "0",
    debitorNrNavision: "",
    name: "Loading...",
    contactPerson: "Loading...",
    contactDetails: "Loading...",
    notes: "Loading...",
    imageUrl: "",
    isArchived: false
};